ibm/service/power/resource_ibm_pi_network_security_group_rule.go

```diff
@@ -0,0 +1,573 @@
+				ForceNew:     true,
+				Required:     true,
+				Type:         schema.TypeString,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			Arg_DestinationPorts: {
+				ConflictsWith: []string{Arg_NetworkSecurityGroupRuleID},
+				Description:   "Destination port ranges.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						Attr_Maximum: {
+							Default:     65535,
+							Description: "The end of the port range, if applicable. If the value is not present then the default value of 65535 will be the maximum port number.",
+							Optional:    true,
+							Type:        schema.TypeInt,
+						},
+						Attr_Minimum: {
+							Default:     1,
+							Description: "The start of the port range, if applicable. If the value is not present then the default value of 1 will be the minimum port number.",
+							Optional:    true,
+							Type:        schema.TypeInt,
+						},
+					},
+				},
+				ForceNew: true,
+				MaxItems: 1,
+				Optional: true,
+				Type:     schema.TypeList,
+			},
+			Arg_NetworkSecurityGroupID: {
+				Description: "The unique identifier of the network security group.",
+				ForceNew:    true,
+				Required:    true,
+				Type:        schema.TypeString,
+			},
+			Arg_NetworkSecurityGroupRuleID: {
+				ConflictsWith: []string{Arg_Action, Arg_DestinationPorts, Arg_Protocol, Arg_Remote, Arg_SourcePorts},
+				Description:   "The network security group rule id to remove.",
+				ForceNew:      true,
+				Optional:      true,
+				Type:          schema.TypeString,
+			},
+			Arg_Protocol: {
+				ConflictsWith: []string{Arg_NetworkSecurityGroupRuleID},
+				Description:   "The protocol of the network traffic.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						Attr_ICMPType: {
+							Description:  "If icmp type, a ICMP packet type affected by ICMP rules and if not present then all types are matched.",
+							Optional:     true,
+							Type:         schema.TypeString,
+							ValidateFunc: validate.ValidateAllowedStringValues([]string{All, DestinationUnreach, Echo, EchoReply, SourceQuench, TimeExceeded}),
+						},
+						Attr_TCPFlags: {
+							Description: "If tcp type, the list of TCP flags and if not present then all flags are matched.",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									Attr_Flag: {
+										Description: "TCP flag.",
+										Required:    true,
+										Type:        schema.TypeString,
+									},
+								},
+							},
+							Optional: true,
+							Type:     schema.TypeList,
+						},
+						Attr_Type: {
+							Description:  "The protocol of the network traffic.",
+							Required:     true,
+							Type:         schema.TypeString,
+							ValidateFunc: validate.ValidateAllowedStringValues([]string{All, ICMP, TCP, UDP}),
+						},
+					},
+				},
+				ForceNew: true,
+				MaxItems: 1,
+				Optional: true,
+				Type:     schema.TypeList,
```

Comment:
Should probably be schema.TypeSet

Reply:
It does not matter here, since only one element is allowed